Observer/SaveOrderMarketingConsent.php

```diff
@@ -0,0 +1,81 @@
+<?php
+
+
+namespace Klaviyo\Reclaim\Observer;
+
+use Exception;
+use Klaviyo\Reclaim\Helper\ScopeSetting;
+use Klaviyo\Reclaim\Helper\Webhook;
+
+use Magento\Framework\Event\Observer;
+use Magento\Framework\Event\ObserverInterface;
+
+
+class SaveOrderMarketingConsent implements ObserverInterface
+{
+    /**
+     * Klaviyo scope setting helper
+     * @var ScopeSetting $klaviyoScopeSetting
+     */
+    protected $_klaviyoScopeSetting;
+
+    /**
+     * @var Webhook $webhookHelper
+     */
+    protected  $_webhookHelper;
+
+    /**
+     * @param Webhook $webhookHelper
+     * @param ScopeSetting $klaviyoScopeSetting
+     */
+    public function __construct(
+        Webhook $webhookHelper,
+        ScopeSetting $klaviyoScopeSetting
+    ) {
+        $this->_webhookHelper = $webhookHelper;
+        $this->_klaviyoScopeSetting = $klaviyoScopeSetting;
+    }
+
+    /**
+     * customer register event handler
+     *
+     * @param Observer $observer
+     * @return void
+     * @throws Exception
+     */
+    public function execute(Observer $observer)
+    {
+        $order = $observer->getEvent()->getOrder();
+        $quote = $observer->getEvent()->getQuote();
+
+        $order->setData("kl_sms_consent", json_encode($quote->getKlSmsConsent()));
+        $order->setData("kl_email_consent", json_encode($quote->getKlEmailConsent()));
+
+        $shippingInfo = $quote->getShippingAddress();
+
+        if (
+            $this->_klaviyoScopeSetting->getWebhookSecret()
+            && $this->_klaviyoScopeSetting->getConsentAtCheckoutSMSIsActive()
+        ) {
+            $data = array(
+                "data" => array(
+                    array(
+                        "customer" => array(
+                            "email" => $quote->getCustomerEmail(),
```

Comment:
i guess you want to add all the other stuff like first name, last name and other pieces of address if possible?

Reply:
we shouldn't need to here, that is all done through app side code, this is really just for list subscriptions, if someone hits this stage, all the other customer properties will come over as well